packages/flutter/test/material/navigation_rail_test.dart

```diff
@@ -0,0 +1,1409 @@
+}
+
+TestSemantics _expectedSemantics() {
+  return TestSemantics.root(
+    children: <TestSemantics>[
+      TestSemantics(
+        textDirection: TextDirection.ltr,
+        children: <TestSemantics>[
+          TestSemantics(
+            flags: <SemanticsFlag>[SemanticsFlag.scopesRoute],
+            children: <TestSemantics>[
+              TestSemantics(
+                flags: <SemanticsFlag>[
+                  SemanticsFlag.isSelected,
+                  SemanticsFlag.isFocusable,
+                ],
+                actions: <SemanticsAction>[SemanticsAction.tap],
+                label: 'Abc\nTab 1 of 5',
+                textDirection: TextDirection.ltr,
+              ),
+              TestSemantics(
+                flags: <SemanticsFlag>[SemanticsFlag.isFocusable],
+                actions: <SemanticsAction>[SemanticsAction.tap],
+                label: 'Def\nTab 2 of 5',
+                textDirection: TextDirection.ltr,
+              ),
+              TestSemantics(
+                flags: <SemanticsFlag>[SemanticsFlag.isFocusable],
+                actions: <SemanticsAction>[SemanticsAction.tap],
+                label: 'Ghi\nTab 3 of 5',
+                textDirection: TextDirection.ltr,
+              ),
+              TestSemantics(
+                flags: <SemanticsFlag>[SemanticsFlag.isFocusable],
+                actions: <SemanticsAction>[SemanticsAction.tap],
+                label: 'Jkl\nTab 4 of 5',
+                textDirection: TextDirection.ltr,
+              ),
+              TestSemantics(
+                flags: <SemanticsFlag>[SemanticsFlag.isFocusable],
+                actions: <SemanticsAction>[SemanticsAction.tap],
+                label: 'Mno\nTab 5 of 5',
+                textDirection: TextDirection.ltr,
+              ),
+              TestSemantics(
+                label: 'body',
+                textDirection: TextDirection.ltr,
+              ),
+            ],
+          ),
+        ],
+      ),
+    ],
+  );
+}
+
+List<NavigationRailDestination> _destinations() {
+  return const <NavigationRailDestination>[
+    NavigationRailDestination(
+      icon: Icon(Icons.favorite_border),
+      activeIcon: Icon(Icons.favorite),
+      label: Text('Abc'),
+    ),
+    NavigationRailDestination(
+      icon: Icon(Icons.bookmark_border),
+      activeIcon: Icon(Icons.bookmark),
+      label: Text('Def'),
+    ),
+    NavigationRailDestination(
+      icon: Icon(Icons.star_border),
+      activeIcon: Icon(Icons.star),
+      label: Text('Ghi'),
+    ),
+    NavigationRailDestination(
+      icon: Icon(Icons.hotel),
+      activeIcon: Icon(Icons.home),
+      label: Text('Jkl'),
+    ),
+    NavigationRailDestination(
+      icon: Icon(Icons.remove_circle),
+      activeIcon: Icon(Icons.add_circle),
+      label: Text('Mno'),
+    ),
+  ];
+}
+
+Future<void> _pumpNavigationRail(
+  WidgetTester tester, {
+  double textScaleFactor,
+  NavigationRail navigationRail,
+}) async {
+  textScaleFactor ??= 1.0;
```

Comment:
This could move into the arguments list itself as a default value.

Reply:
Done